### ISSUE TITLE
suggestion for batch of attacks?

### ISSUE THREAD

**Issue by gobbedy** [bethgelab/foolbox#240]
I just started coding with foolbox, and hoping for a bit of advice.

I'd like to use adversarial attacks themselves for regularization.

So I need to run the same attack on many samples (images) in parallel.

In particular, I'm hoping to use Carlini_Wagner attack. I just now built my code using a for loop, like this:

```python
            images = <grab a batch of images>
            for idx, image in enumerate(images):
                adversarial = attack(image, labels[idx])
                ...
```

But a for loop is inefficient, and I'm noticing that the Carlini_Wagner attack is quite slow.

Any suggestions on how I might run the same attack in parallel efficiently?

I'm assuming there's no magic pill, but I thought I'd ask :)

**Comment by wielandbrendel** [comment]
We are aware of this issue, and we are currently implementing a batch-processing mode for all adversarial attacks that work in parallel in many samples (so exactly what you want). This feature should be ready within the next two weeks. Please be patient :).

**Comment by gobbedy** [comment]
Omg. Of course you're already working on it. Why am I not surprised :)

**Comment by gobbedy** [comment]
Closing since I just realized it's a duplicate of issue #208.
